arangod/IResearch/ViewSnapshot.cpp

```diff
@@ -163,30 +167,26 @@ void syncViewSnapshot(ViewSnapshot& snapshot, std::string_view name) {
 
 ViewSnapshot* makeViewSnapshot(transaction::Methods& trx, void const* key,
                                bool sync, std::string_view name,
-                               ViewSnapshot::Links&& links) noexcept {
+                               ViewSnapshot::Links&& links) {
   TRI_ASSERT(trx.state());
   auto& state = *(trx.state());
   TRI_ASSERT(state.cookie(key) == nullptr);
+
+  for (auto const& link : links) {
+    if (link->hasFailed()) {
```

Comment:
link can be null

Reply:
if (!link) {
      LOG_TOPIC("fffff", WARN, TOPIC)
          << "failed to lock a link for view '" << name;
      return false;
    }